extensions/notebook/src/dialog/managePackages/managePackagesDialogModel.ts

```diff
@@ -0,0 +1,281 @@
+		return this.options.defaultLocation || this.targetLocations[0];
+	}
+
+	/**
+	 * Returns the default location
+	 */
+	public get defaultProviderId(): string {
+		return this.options.defaultProviderId || Array.from(this.packageManageProviders.keys())[0];
+	}
+
+	/**
+	 * Loads the provider cache
+	 */
+	private async loadCaches(): Promise<void> {
+		if (this.packageManageProviders) {
+			let keyArray = Array.from(this.packageManageProviders.keys());
+			for (let index = 0; index < keyArray.length; index++) {
+				const element = this.packageManageProviders.get(keyArray[index]);
+				if (await element.canUseProvider()) {
+					if (!this._locations.has(element.packageTarget.location)) {
+						this._locations.add(element.packageTarget.location);
+					}
+					if (!this._packageTypes.has(element.packageTarget.location)) {
+						this._packageTypes.set(element.packageTarget.location, []);
+					}
+					this._packageTypes.get(element.packageTarget.location).push(element);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Returns a map of providerId to package types for given location
+	 */
+	public getPackageTypes(targetLocation?: string): ProviderPackageType[] {
+		targetLocation = targetLocation || this.defaultLocation;
+		let providers = this._packageTypes.get(targetLocation);
+		return providers.map(x => {
+			return {
+				providerId: x.providerId,
+				packageType: x.packageTarget.packageType
+			};
+		});
+	}
+
+	/**
+	 * Returns a map of providerId to package types for given location
+	 */
+	public getDefaultPackageType(): ProviderPackageType {
+		let defaultProviderId = this.defaultProviderId;
+		let packageTypes = this.getPackageTypes();
+		return packageTypes.find(x => x.providerId === defaultProviderId);
+	}
+
+	/**
+	 * returns the list of packages for current provider
+	 */
+	public async listPackages(): Promise<IPackageDetails[]> {
+		let provider = this.currentPackageManageProvider;
+		if (provider) {
+			return await provider.listPackages();
+		} else {
+			throw new Error('Current Provider is not set');
+		}
+	}
+
+	/**
+	 * Changes the current provider
+	 */
+	public changeProvider(providerId: string): void {
+		if (this._packageManageProviders.has(providerId)) {
+			this._currentProvider = providerId;
+		} else {
+			throw Error(`Invalid package type ${providerId}`);
+		}
+	}
+
+	/**
+	 * Installs given packages using current provider
+	 * @param packages Packages to install
+	 */
+	public async installPackages(packages: IPackageDetails[]): Promise<void> {
+		let provider = this.currentPackageManageProvider;
+		if (provider) {
+			await provider.installPackages(packages, false);
+		} else {
+			throw new Error('Current Provider is not set');
+		}
+	}
+
+	/**
+	 * Returns the location title for current provider
+	 */
+	public async getLocationTitle(): Promise<string> {
+		let provider = this.currentPackageManageProvider;
+		if (provider) {
+			return await provider.getLocationTitle();
+		}
+		return Promise.resolve(undefined);
```

Comment:
You can just return here. And the return type should be `Promise<string | undefined>` to follow strict null check guidelines